viam-cartographer_test.go

```diff
@@ -127,6 +133,53 @@ func TestNew(t *testing.T) {
 	})
 
 	testhelper.ClearDirectory(t, dataDir)
+
+	t.Run("Successful creation of cartographer slam service in localization mode", func(t *testing.T) {
+		grpcServer, port := setupTestGRPCServer(t)
+
+		attrCfg := &vcConfig.Config{
+			Sensors:       []string{},
+			ConfigParams:  map[string]string{"mode": "2d"},
+			DataDirectory: dataDir,
+			Port:          "localhost:" + strconv.Itoa(port),
+			UseLiveData:   &_false,
+			MapRateSec:    &_zeroInt,
+		}
+
+		svc, err := testhelper.CreateSLAMService(t, attrCfg, logger, false, testExecutableName)
+		test.That(t, err, test.ShouldBeNil)
+
+		grpcServer.Stop()
+		timestamp1, _ = svc.GetLatestMapInfo(context.Background())
```

Comment:
add in the nil error check here. Even though it appears redundant, its worth doing incase the behavior changes